src/charts/series-chart.js

```diff
@@ -26,74 +26,82 @@ import {utils} from './utils';
-    }
+        this._keySort = (a, b) => d3.ascending(this.keyAccessor()(a), this.keyAccessor()(b));
+
+        this._charts = {};
+        this._chartFunction = lineChart;
+        this._chartGroup = chartGroup;
+        this._seriesAccessor = undefined;
+        this._seriesSort = d3.ascending;
+        this._valueSort = this._keySort;
 
-    var _charts = {};
-    var _chartFunction = lineChart;
-    var _seriesAccessor;
-    var _seriesSort = d3.ascending;
-    var _valueSort = keySort;
+        this._mandatoryAttributes().push('seriesAccessor', 'chart');
+        this.shareColors(true);
 
-    _chart._mandatoryAttributes().push('seriesAccessor', 'chart');
-    _chart.shareColors(true);
+        // ES6: consider creating a method _compose that calls super.compose and make overridden compose to throw exception
+        // make compose private
+        this._compose = this.compose;
+        delete this.compose;
+    }
 
-    _chart._preprocessData = function () {
-        var keep = [];
-        var childrenChanged;
-        var nester = d3.nest().key(_seriesAccessor);
-        if (_seriesSort) {
-            nester.sortKeys(_seriesSort);
+    _preprocessData () {
+        const keep = [];
+        let childrenChanged;
+        const nester = d3.nest().key(this._seriesAccessor);
+        if (this._seriesSort) {
+            nester.sortKeys(this._seriesSort);
         }
-        if (_valueSort) {
-            nester.sortValues(_valueSort);
+        if (this._valueSort) {
+            nester.sortValues(this._valueSort);
         }
-        var nesting = nester.entries(_chart.data());
-        var children =
-            nesting.map(function (sub, i) {
-                var subChart = _charts[sub.key] || _chartFunction.call(_chart, _chart, chartGroup, sub.key, i);
-                if (!_charts[sub.key]) {
+        const nesting = nester.entries(this.data());
+        const children =
+            nesting.map((sub, i) => {
+                const subChart = this._charts[sub.key] || this._chartFunction.call(this, this, this._chartGroup , sub.key, i);
+                if (!this._charts[sub.key]) {
                     childrenChanged = true;
                 }
-                _charts[sub.key] = subChart;
+                this._charts[sub.key] = subChart;
                 keep.push(sub.key);
                 return subChart
-                    .dimension(_chart.dimension())
+                    .dimension(this.dimension())
                     .group({
                         all: typeof sub.values === 'function' ? sub.values : utils.constant(sub.values)
                     }, sub.key)
-                    .keyAccessor(_chart.keyAccessor())
-                    .valueAccessor(_chart.valueAccessor())
+                    .keyAccessor(this.keyAccessor())
+                    .valueAccessor(this.valueAccessor())
                     .brushOn(false);
             });
+        // ES6: do we add removal function in composite chart?
```

Comment:
Although it's a great idea, it's not really an ES6 issue, so let's deal with it later. 

As always, adding features requires thinking carefully about impact, as well as tedious but valuable test-writing.

For now, I don't think `chart` is necessary here. (much easier discussion!)

Reply:
I agree, let us drop this.